### PR TITLE
`azurerm_storage_account` - Add 403 (previously only 401) as a valid status code for lacking permissions to list keys

### LINKED ISSUE #19622: Importing a storage account into my state exposes the access keys and requires list permission for the keys
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.6

### AzureRM Provider Version

3.32.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "storage_account" {
  name                     = var.name
  resource_group_name      = var.rg 
  location                 = var.location
  account_tier             = "Standard"
  account_replication_type = "RAGRS"
}
```

### Debug Output/Panic Output

```shell
Error: listing Keys for Storage Account: (Name "<Storage Account ID>" / Resource Group "<Resource Group Name>"): storage.AccountsClient#ListKeys: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client '<Client Email>' with object id '<User ID>' does not have authorization to perform action 'Microsoft.Storage/storageAccounts/listKeys/action' over scope '<Resource ID>' or the scope is invalid. If access was recently granted, please refresh your credentials."
```

### Expected Behaviour

The expected behaviour was that the storage account is imported into the state.

### Actual Behaviour

While importing the storage account, the import fails because of missing permissions to list the keys (see error message in 'Debug Output')

### Steps to Reproduce

- Create a storage account in the portal
- Write a small Terraform file for a storage account like this:
```terraform
resource "azurerm_storage_account" "storage_account" {
     name                     = var.name
     resource_group_name      = var.rg
     location                 = var.location
     account_tier             = "Staard"
     account_replication_type = "RAGRS"
}
```
- Terraform init in your directory
- Try to import the storage account into your state with and without list permissions on the access keys:
`terraform import azurerm_storage_account <Resource ID>`

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by magodo**
Per #19622, the API now returns 403 (at least for this case) when users have no permission:

> Status=403 Code="AuthorizationFailed" Message="The client '<Client Email>' with object id '<User ID>' does not have authorization to perform action 'Microsoft.Storage/storageAccounts/listKeys/action' over scope '<Resource ID>' or the scope is invalid. If access was recently granted, please refresh your credentials."

This PR adds 403 (previously only 401) as a valid status code for lacking permissions to list keys. 

Fix #19622

### CODE REVIEW

File: internal/services/storage/storage_account_resource.go [hunk 1]
```diff
@@ -1900,7 +1900,7 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 		if e, ok := err.(azautorest.DetailedError); ok {
 			if status, ok := e.StatusCode.(int); ok {
 				hasWriteLock = status == http.StatusConflict
-				doesntHavePermissions = status == http.StatusUnauthorized
+				doesntHavePermissions = (status == http.StatusUnauthorized || status == http.StatusForbidden)
```
Comment (tombuildsstuff) [inline internal/services/storage/storage_account_resource.go:1903]: @magodo it'd be worth spinning one up to check, but I have a feeling that a 403 is a valid error here for the Resource (but wouldn't be valid for the Data Source, where we allow for less permissive permissions)?

From memory a 401 indicates differently that access keys are disabled on the storage account too?

Comment (magodo) [reply internal/services/storage/storage_account_resource.go:1903]: My local test shows that if `shared_access_key_enabled = false`, the `listKeys` call still succeeds, while the follow up calls to service like queue with shared key auth will fail with 403, complaining "Key based authentication is not permitted on this storage account.". Using `storage_use_azuread = true` in the provider block can move on.

Meanwhile, I've created a new app which is only assigned following roles:

- Microsoft.Resources/subscriptions/resourceGroups/read
- Microsoft.Storage/storageAccounts/read

Using this SP to read the storage account via the provider returns 403:

> │ Error: listing Keys for Storage Account: (Name "unlikely24exst2acctv9ipk" / Resource Group "acctestRG-storage-220816112105435513"): storage.AccountsClient#ListKeys: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'xxxx' with object id 'xxxx' does not have authorization to perform action 'Microsoft.Storage/storageAccounts/listKeys/action' over scope '/subscriptions/xxxx/resourceGroups/acctestRG-storage-220816112105435513/providers/Microsoft.Storage/storageAccounts/unlikely24exst2acctv9ipk' or the scope is invalid. If access was recently granted, please refresh your credentials."

In the end, I didn't figure out when do we get 401.

Comment (tombuildsstuff) [reply internal/services/storage/storage_account_resource.go:1903]: sounds reasonable to me, let's leave the 401 thing in for now 👍

**Review by katbyte — APPROVED** [review]
LGTM 🌊

**Pull request merged**